homeassistant/components/bmw_connected_drive/__init__.py

```diff
@@ -167,7 +134,7 @@ def _update_all() -> None:
             Platform.NOTIFY,
             DOMAIN,
             {CONF_NAME: DOMAIN},
-            hass.data[DOMAIN][DATA_HASS_CONFIG],
+            dict(entry.data),
```

Comment:
This is incorrect. We still need to pass the original whole config to `async_load_platform` as last argument. So please keep `async_setup` above for that purpose of storing the config in hass.data so we can access it here as before.

Reply:
I can add that back, but I don't understand why we need it.

The notify service gets set up only via `hass.data` ([notify.py#L36-L38](https://github.com/home-assistant/core/blob/ef7a3adaf4577e94443cb13f335d38bf1f13f079/homeassistant/components/bmw_connected_drive/notify.py#L36-L38)) and I have verified that it works correctly.

Reply:
If the integration we set up, eg notify, has other dependencies, those may need data from the config. It doesn't matter if that's true right now. We need to follow the signature of `async_load_platform` and call it correctly.

Reply:
Ok, I have added it back.

Just for my understanding: This is only needed as long as the notify platform still uses `discovery.async_load_platform()`, correct?
Just asking because I would have similar issues in #67003.